### PR TITLE
Further refactoring for PEP8 names.

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #19

### CODE REVIEW

**Comment by MichaelClerx** [comment]
A few more bits and pieces. I think we can consider #19 closed after this one

**Comment by MichaelClerx** [comment]
This one's not as scary as the previous one :D

File: fc/parsing/CompactSyntaxParser.py [hunk 1]
```diff
@@ -204,9 +204,9 @@ def OperatorOperands(self):
                 except StopIteration:
                     break
 
-        def Operator(self, operator):
-            """Get the MathML element for the given operator."""
-            return getattr(M, self.OP_MAP[operator])
+        # def operator(self, operator):
```
Comment (jonc125) [inline fc/parsing/CompactSyntaxParser.py]: Can we just delete this, and the definition of `OP_MAP`?

File: fc/parsing/CompactSyntaxParser.py [hunk 2]
```diff
@@ -1025,102 +1027,102 @@ class CompactSyntaxParser(object):
...
-    nullValue = p.Group(MakeKw('null')).setName('Null').setParseAction(Actions.Symbol('null'))
-    defaultValue = p.Group(MakeKw('default')).setName('Default').setParseAction(Actions.Symbol('defaultParameter'))
-    stringValue = quotedString.copy().setName('String').setParseAction(Actions.Symbol('string'))
+    null_value = p.Group(make_kw('null')).setName('Null').setParseAction(Actions.Symbol('null'))
+    default_value = p.Group(make_kw('default')).setName('Default').setParseAction(Actions.Symbol('defaultParameter'))
+    string_value = quoted_string.copy().setName('String').setParseAction(Actions.Symbol('string'))
 
     # Recognised MathML operators
-    mathmlOperators = set('''quotient rem max min root xor abs floor ceiling exp ln log
+    mathml_operators = set('''quotient rem max min root xor abs floor ceiling exp ln log
```
Comment (jonc125) [inline fc/parsing/CompactSyntaxParser.py]: Tidy up the indentation on the next few lines?

File: fc/parsing/CompactSyntaxParser.py [hunk 3]
```diff
@@ -1025,102 +1027,102 @@ class CompactSyntaxParser(object):
...
                              sin cos tan   sec csc cot   sinh cosh tanh   sech csch coth
                              arcsin arccos arctan   arccosh arccot arccoth
                              arccsc arccsch arcsec   arcsech arcsinh arctanh'''.split())
 
     # Wrapping MathML operators into lambdas
-    mathmlOperator = (p.oneOf('^ * / + - not == != <= >= < > && ||') |
-                      p.Combine('MathML:' + p.oneOf(' '.join(mathmlOperators))))
-    wrap = p.Group(p.Suppress('@') - Adjacent(p.Word(p.nums)) + Adjacent(colon) + mathmlOperator
+    mathml_operator = (p.oneOf('^ * / + - not == != <= >= < > && ||') |
+                      p.Combine('MathML:' + p.oneOf(' '.join(mathml_operators))))
```
Comment (jonc125) [inline fc/parsing/CompactSyntaxParser.py]: Also here?

File: fc/parsing/CompactSyntaxParser.py [hunk 4]
```diff
@@ -1445,21 +1447,21 @@ def try_parse(self, callable, *args, **kwargs):
                                % (int(self._stack_depth_factor * self._original_stack_limit),))
         return r
 
-    # def ParseFile(self, filename, xmlGenerator=None):
+    # def parse_file(self, filename, xml_generator=None):
```
Comment (jonc125) [inline fc/parsing/CompactSyntaxParser.py]: We can probably delete this function?

File: test/test_CompactSyntaxParser.py [hunk 5]
```diff
@@ -125,7 +125,7 @@ def test_parsing_simple_expressions(self):
                           [[['1', '<', '2'], ['3', '+', '4'], ['5', '*', '6']]])
 
     # XML-Only method?
-    # def testParsingTrace(self):
+    # def test_parsing_trace(self):
```
Comment (jonc125) [inline test/test_CompactSyntaxParser.py]: Yeah, this is XML-only and can be removed. The Python implementation can trace naturally using the stack, so doesn't need special support!

File: test/test_CompactSyntaxParser.py [hunk 6]
```diff
@@ -833,15 +833,15 @@ def test_parsing_post_processing(self):
 }
 """
         out = [[[['a'], [['check', ['sim:result']]]], [['a', '>', '5']]]]
-        self.assertParses(csp.postProcessing, mls, out)
+        self.assertParses(csp.post_processing, mls, out)
 
     '''
     def test_parsing_full_protocols(self):
```
Comment (jonc125) [inline test/test_CompactSyntaxParser.py]: I reckon we can delete this test too.

File: fc/parsing/CompactSyntaxParser.py (hunk 1 above)
Comment (MichaelClerx) [reply fc/parsing/CompactSyntaxParser.py]: ok!

File: fc/parsing/CompactSyntaxParser.py (hunk 2 above)
Comment (MichaelClerx) [reply fc/parsing/CompactSyntaxParser.py]: will try!

File: fc/parsing/CompactSyntaxParser.py (hunk 3 above)
Comment (MichaelClerx) [reply fc/parsing/CompactSyntaxParser.py]: will try1

File: fc/parsing/CompactSyntaxParser.py (hunk 4 above)
Comment (MichaelClerx) [reply fc/parsing/CompactSyntaxParser.py]: ok!

File: test/test_CompactSyntaxParser.py (hunk 5 above)
Comment (MichaelClerx) [reply test/test_CompactSyntaxParser.py]: Do we still need the `class Trace` and other tracing methods then?

File: test/test_CompactSyntaxParser.py (hunk 6 above)
Comment (MichaelClerx) [reply test/test_CompactSyntaxParser.py]: done!

File: test/test_CompactSyntaxParser.py (hunk 5 above)
Comment (jonc125) [reply test/test_CompactSyntaxParser.py]: Ah, sorry, I misremembered. There is still something that happens in the Python implementation with this - it sets `wrapped_expr.trace = True`, which triggers a call to `Locatable.trace_value` when the expression is evaluated. It's a debugging aid, though probably isn't used much!

You could enhance the test with a check like this?
```python
action = grammar.parseString('var?', parseAll=True)
assert action.expr().trace

action = grammar.parseString('var', parseAll=True)
assert not action.expr().trace
```

Comment (MichaelClerx) [reply test/test_CompactSyntaxParser.py]: So what would I do with the PROTO_NS bit?

```
    def testParsingTrace(self):
       trace = {'{%s}trace' % CSP.PROTO_NS: '1'}
       self.assertParses(csp.expr, '1?', [['1']])
       self.assertParses(csp.expr, 'var?', [['var']])
       self.assertParses(csp.expr, '(1 + a)?', [[['1', '+', 'a']]])
       self.assertParses(csp.expr, '1 + a?', [['1', '+', ['a']]])
```

Comment (jonc125) [reply test/test_CompactSyntaxParser.py]: That bit can go - it was just checking that the XML conversion added the attribute.

Comment (MichaelClerx) [reply test/test_CompactSyntaxParser.py]:
```
        action = csp.expr.parseString('var?', parseAll=True)
        print(action)
        print(type(action))
        assert action._expr().trace
```
I've tried this but that returns:
```
[Trace[Variable[var]]]
<class 'pyparsing.ParseResults'>
```
and then
```
    assert action._expr().trace
E   TypeError: 'str' object is not callable
```

Comment (jonc125) [reply test/test_CompactSyntaxParser.py]: Maybe you need `action[0].expr().trace` then? Looks like the `Action` class is wrapped from the outer `[]`

**Pull request merged**